pkg/netutils/netutils_test.go

```diff
@@ -0,0 +1,1026 @@
+
+	resultJSON, ok := cachedInfo["result"].(map[string]interface{})
+	ExpectWithOffset(1, ok).To(BeTrue())
+
+	resultByte, err := json.Marshal(resultJSON)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	return resultByte
+}
+
+func test020ResultHasIPv4DefaultRoute(data []byte) bool {
+	resultRaw, err := types020.NewResult(data)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	result, err := types020.GetResult(resultRaw)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	for _, r := range result.IP4.Routes {
+		if r.Dst.String() == "0.0.0.0/0" {
+			return true
+		}
+	}
+	return false
+}
+
+func test020ResultHasIPv6DefaultRoute(data []byte) bool {
+	resultRaw, err := types020.NewResult(data)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	result, err := types020.GetResult(resultRaw)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	for _, r := range result.IP6.Routes {
+		if r.Dst.String() == "::/0" {
+			return true
+		}
+	}
+	return false
+}
+
+func testResultHasIPv4DefaultRoute(data []byte) bool {
+	resultRaw, err := current.NewResult(data)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	result, err := current.GetResult(resultRaw)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	for _, r := range result.Routes {
+		if r.Dst.String() == "0.0.0.0/0" {
+			return true
+		}
+	}
+	return false
+}
+
+func testResultHasIPv6DefaultRoute(data []byte) bool {
+	resultRaw, err := current.NewResult(data)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	result, err := current.GetResult(resultRaw)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	for _, r := range result.Routes {
+		if r.Dst.String() == "::/0" {
+			return true
+		}
+	}
+	return false
+}
+
+var _ = Describe("netutil netlink function testing", func() {
+	const IFNAME string = "dummy0"
+	var IFMAC net.HardwareAddr = net.HardwareAddr([]byte{0x02, 0x66, 0x7d, 0xe3, 0x14, 0x1c})
+	var originalNS ns.NetNS
+	var targetNS ns.NetNS
+
+	BeforeEach(func() {
+		// Create a new NetNS so we don't modify the host
+		var err error
+		originalNS, err = testutils.NewNS()
+		Expect(err).NotTo(HaveOccurred())
+
+		targetNS, err = testutils.NewNS()
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(targetNS.Do(func(ns.NetNS) error {
+			defer GinkgoRecover()
+
+			Expect(netlink.LinkAdd(&netlink.Dummy{
+				LinkAttrs: netlink.LinkAttrs{
+					Name: IFNAME,
+					HardwareAddr: IFMAC,
+					Index: 10,
+				},
+			})).Should(Succeed())
+
+			_, err = netlink.LinkByName(IFNAME)
+			Expect(err).NotTo(HaveOccurred())
+
+			return nil
+		})).Should(Succeed())
+	})
+
+	AfterEach(func() {
+		Expect(originalNS.Close()).To(Succeed())
+	})
+
+	Context("test DeleteDefaultGW", func() {
+		It("verify default gateway is removed", func() {
+			Expect(targetNS.Do(func(ns.NetNS) error {
+				defer GinkgoRecover()
+				link, err := netlink.LinkByName(IFNAME)
+				Expect(err).NotTo(HaveOccurred())
+				err = netlink.LinkSetUp(link)
+				Expect(err).NotTo(HaveOccurred())
```

Comment:
optional nit: these could be inlined. Sorry for having missed it in the first pass.